src/utils/installPackage.js

```diff
@@ -13,12 +14,18 @@ async function install(dir, modules, installPeers = true) {
       cwd: location ? path.dirname(location) : dir
     };
 
-    if (location && path.basename(location) === 'yarn.lock') {
-      install = spawn('yarn', ['add', ...modules, '--dev'], options);
-    } else {
-      install = spawn('npm', ['install', ...modules, '--save-dev'], options);
+    let args = ['add', ...modules];
+    if (saveDev) {
+      args.push('-D');
     }
 
+    let command = 'npm';
+    if (commandExists('yarn')) {
```

Comment:
Better preserve the old logic of `location && path.basename(location) === 'yarn.lock'`. Just because `yarn` is installed, doesn't mean it is being used.

Reply:
Interesting... is there a case where you wouldn’t want to use yarn if you already have it?

Reply:
Interesting... is there a case where you wouldn’t want to use yarn if you already have it?

Reply:
Sure there is! I have yarn and I don't use it for every project. Specially when working with a team, everyone must use the same package manager for that project.

Reply:
Got it :)

We should keep both though, right? So we should check for the yarn lock file **AND** the yarn command, that way if they have a yarn lock but no yarn they can still use npm. Or is that also not necessary?

Reply:
I see no need for keeping both checks, but I see no harm as well. Your call.

Reply:
+1 for keeping the `yarn.lock` check. I'm OK with both, but the lock check is going to be an indication of what the team / project is using.